### PR TITLE
Joomla 4 Compatability

### PR DESCRIPTION
**Pull request by ankush-maherwal**

### CODE REVIEW

File: code/plugins/transfirst/transfirst.php [hunk 1]
```diff
@@ -7,12 +7,18 @@
  * @license    GNU General Public License version 2 or later.
  */
 defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Filesystem\File;
+use Joomla\CMS\Language\Text;
+
 jimport('joomla.filesystem.file');
```
Comment (manojLondhe) [inline code/plugins/transfirst/transfirst.php]: remove unused jimports

File: code/plugins/payumoney/payumoney/helper.php [hunk 2]
```diff
@@ -6,6 +6,10 @@
 
 defined('_JEXEC') or die('Restricted access');
 
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Log\Log;
+
 jimport('joomla.html.html');
```
Comment (manojLondhe) [inline code/plugins/payumoney/payumoney/helper.php]: remove unused jimports

File: code/plugins/payumoney/payumoney.php [hunk 3]
```diff
@@ -6,11 +6,16 @@
 
 // No direct access
 defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Filesystem\File;
+
 jimport('joomla.plugin.plugin');
```
Comment (manojLondhe) [inline code/plugins/payumoney/payumoney.php]: remove unused jimports

File: code/plugins/payu/payu/helper.php [hunk 4]
```diff
@@ -8,6 +8,10 @@
  */
 defined('_JEXEC') or die('Restricted access');
 
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Log\Log;
+
 jimport('joomla.html.html');
```
Comment (manojLondhe) [inline code/plugins/payu/payu/helper.php]: remove unused jimports

File: code/plugins/amazon/amazon/tmpl/default.php [hunk 5]
```diff
@@ -7,6 +7,9 @@
  */
 
 defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Language\Text;
+
 if(JVERSION >='1.6.0')
```
Comment (manojLondhe) [inline code/plugins/amazon/amazon/tmpl/default.php]: see if this can be removed

File: code/plugins/transfirst/transfirst.php (hunk 1 above)
Comment (ankush-maherwal) [reply code/plugins/transfirst/transfirst.php]: done

File: code/plugins/payumoney/payumoney.php (hunk 3 above)
Comment (ankush-maherwal) [reply code/plugins/payumoney/payumoney.php]: done

File: code/plugins/amazon/amazon/tmpl/default.php (hunk 5 above)
Comment (ankush-maherwal) [reply code/plugins/amazon/amazon/tmpl/default.php]: done

**Pull request merged**